### PR TITLE
support compaction filter

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang @disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

**Comment by BusyJay** [comment]
cc pingcap/tikv#904

File: test/test_compaction_filter.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    let samples = vec![
+        (b"key1".to_vec(), b"value1".to_vec()),
+        (b"key2".to_vec(), b"value2".to_vec()),
+    ];
+    for &(ref k, ref v) in &samples {
+        db.put(k, v).unwrap();
+        assert_eq!(v.as_slice(), &*db.get(k).unwrap().unwrap());
+    }
+    {
+        let _snap = db.snapshot();
```
Comment (zhangjinpeng1987) [inline test/test_compaction_filter.rs]: I suggest you'd better add some comments for this key point.

**Comment by siddontang** [comment]
LGTM

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**